test/90_1_prm/test_appendix_g_prm.rb

```diff
@@ -330,6 +333,80 @@ def check_lpd(prototypes_base)
     end
   end
 
+  # Check lighting occ sensor
+  #
+  # @param prototypes_base [Hash] Baseline prototypes
+  def check_light_occ_sensor(prototypes,prototypes_base)
+    light_sch = {}
+    prototypes.each do |prototype, model_proto|
+      building_type, template, climate_zone, mod = prototype
+      # Define name of spaces used for verification
+      run_id = "#{building_type}_#{template}_#{climate_zone}_#{mod}"
+      space_name = JSON.parse(File.read("#{@@json_dir}/light_occ_sensor.json"))[run_id]
+      
+      # Get lighting schedule in prototype model
+      light_sch_model = {}
+      model_proto.getLightss.sort.each do |lgts|
+        light_sch_model_lgts = {}
+        lgts.schedule.get.to_ScheduleRuleset.get.scheduleRules.each do |week_rule|
+          light_sch_model_week_rule = {}
+          day_rule = week_rule.daySchedule()
+          times = day_rule.times()
+          times.each do |time|
+            light_sch_model_week_rule[time.to_s] = day_rule.getValue(time)
+          end
+          light_sch_model_lgts[week_rule.name.to_s] = light_sch_model_week_rule
+        end
+        light_sch_model[lgts.name.to_s] = light_sch_model_lgts
+      end
+      light_sch[run_id] = light_sch_model
+    end
+
+    light_sch_base = {}
+    prototypes_base.each do |prototype, model_baseline|
+      building_type, template, climate_zone, mod = prototype
+      # Define name of spaces used for verification
+      run_id = "#{building_type}_#{template}_#{climate_zone}_#{mod}"
+      space_name = JSON.parse(File.read("#{@@json_dir}/light_occ_sensor.json"))[run_id]
+    
+      # Get lighting schedule in baseline model
+      model_baseline.getSpaceTypes.sort.each do |space_type|
+        light_sch_model_base = {}
+        space_type.lights.sort.each do |lgts|
+          light_sch_model_lgts_base = {}
+          light_sch_model_lgts_base['space_type'] = space_type.standardsSpaceType.to_s
+          lgts.schedule.get.to_ScheduleRuleset.get.scheduleRules.each do |week_rule|
+            light_sch_model_week_rule_base = {}
+            day_rule = week_rule.daySchedule()
+            times = day_rule.times()
+            times.each do |time|
+              light_sch_model_week_rule_base[time.to_s] = day_rule.getValue(time)
+            end
+            light_sch_model_lgts_base[week_rule.name.to_s] = light_sch_model_week_rule_base
+          end
+          light_sch_model_base[lgts.name.to_s] = light_sch_model_lgts_base
+        end
```

Comment:
It might be more concise to have this section and the same one above in a function and just call the function from this test.

Reply:
The two loops have some differences. Currently, I still keep two loops.